Kernel/Devices/MBRPartitionTable.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <AK/RetainPtr.h>
+#include <AK/Vector.h>
+#include <Kernel/Devices/DiskDevice.h>
+#include <Kernel/Devices/DiskPartition.h>
+
+#define MBR_SIGNATURE 0xaa55
+
+struct MBRPartitionEntry {
+    byte status;
+    byte chs1[3];
+    byte type;
+    byte chs2[3];
+    dword offset;
+    dword length;
+} __attribute__((packed));
+
+struct MBRPartitionHeader {
+    byte code1[218];
+    word ts_zero;
+    byte ts_drive, ts_seconds, ts_minutes, ts_hours;
+    byte code2[216];
+    dword disk_signature;
+    word disk_signature_zero;
+    MBRPartitionEntry entry[4];
+    word mbr_signature;
+} __attribute__((packed));
+
+class MBRPartitionTable : public Retainable<MBRPartitionTable> {
```

Comment:
Since we never free these objects, let's use `AK_MAKE_ETERNAL` to allocate them using `kmalloc_eternal()` (a special allocator that doesn't worry about being able to free):

```cpp
class MBRPartitionTable : public Retainable<MBRPartitionTable> {
    AK_MAKE_ETERNAL
public:
    ....
```